### ISSUE TITLE
[feat] rh-footer should be able to display global footer only

### ISSUE THREAD

**Issue by heyMP** [RedHat-UX/red-hat-design-system#371]
### Description

We missed this feature in the initial release of rh-footer but there are some cases where only the global footer that contains the legal links is needed.  Currently we don't have a method of doing that easily.  This is because the 'header' section of the footer has fallback content if no slots are provided and because the 'main' section of the component has height due to grid gap.  This means that even if there is no content for 'main' and 'header' slots defined by the user, they will still see those sections.

### Suggested solution

We should use the SlotController to hide sections that don't contain user provided content for the 'header', 'main', and 'footer' sections of the rh-footer component.

### Additional context

https://deploy-preview-341--red-hat-design-system.netlify.app/components/footer/#global-footer

![Screen Shot 2022-06-07 at 9 34 12 AM](https://user-images.githubusercontent.com/3428964/172393639-f5427b5c-0926-4afb-b8d2-a0b138890a49.png)

**Comment by bennypowers** [comment]
What if 
```html
<!-- happy path -->
<rh-footer>
  ... insert normal footer content here
  <rh-global-footer slot="footer">...insert global content here</rh-global-footer>
</rh-footer>

<!-- global only -->
<rh-global-footer>...insert global content here</rh-global-footer>
```

**Comment by heyMP** [comment]
Could do that too.  Normally I would go with a dedicated component but it looks like we're leaning towards using slots to do more of the heavy lifting. 

Example: https://github.com/RedHat-UX/red-hat-design-system/issues/281#issuecomment-1134698422

**Comment by bennypowers** [comment]
yeah i still think that's the way to go, generally speaking

but in this particular case the request was for *only* the global part
so we can save some bytes on those pages by vending a separate component

**Comment by heyMP** [comment]
@bennypowers good point about the saving bytes consideration.  We'll also have to make sure that we do the following:

- [ ] Ensure that the `rh-footer-global` is not bundled with `rh-footer.js`

**Comment by bennypowers** [comment]
will be a la carte just like the rest of the lib if we
```js
import '@rhds/elements/rh-footer/rh-global-footer.js';
```

Will it need lightDOM styles?

**Comment by heyMP** [comment]
It will need lightdom styles since there are nested lists.  We have two options.

1. Create a lightdom CSS file for `rh-global-footer.js` that contains relevant styles for css.  Downside is that we would have to maintain the same style definitions in two separate CSS files. One in `rh-global-footer-lightdom.css` and one in `rh-footer-lightdom.css`
2. We have rh-global-footer rely on `rh-footer-lightdom.css`.  Downside would be that global footer would some unused styles.
3. We could have `rh-footer` require both `rh-footer-lightdom.css` and `rh-global-footer-lightdom.css`.  😕

**Comment by bennypowers** [comment]
1 and 3 hurt my heart.

I think 2 (have global footer rely on the main footer-lightdom sheet) is best because, as your friend and mine Frank Taylor pointed out, excess global css is not as hard as excess JS. If it's possible to minimize the amount of excess (by sharing selectors, for example), that would be ideal

**Comment by bennypowers** [comment]
Closed by #406
